src/ai/defaultai.cc

```diff
@@ -3341,70 +3344,219 @@ void DefaultAI::diplomacy_actions(const Time& gametime) {
+					// Replace plan to leave with accepting invite, which also leaves old team
+					plan_priority = 0;
+				}
 			}
 
-			game().send_player_diplomacy(pda.other,
-			                             (pda.action == Widelands::DiplomacyAction::kInvite ?
-                                          (accept ? Widelands::DiplomacyAction::kAcceptInvite :
-                                                    Widelands::DiplomacyAction::kRefuseInvite) :
-                                          (accept ? Widelands::DiplomacyAction::kAcceptJoin :
-                                                    Widelands::DiplomacyAction::kRefuseJoin)),
-			                             pda.sender);
-			verb_log_dbg_time(
-			   gametime, "AI Diplomacy: Player(%d)%s %s %s player (%d) with diploscore %d%s\n",
-			   static_cast<unsigned int>(pda.other), myts_s.c_str(), accept ? "accepts" : "denies",
-			   pda.action == Widelands::DiplomacyAction::kInvite ? "the invitation of" : "to join",
-			   static_cast<unsigned int>(pda.sender), diploscore, other_team_score_str.c_str());
+			accept = accept && (planned_action == kNoAction || plan_priority < priority);
 
-			if (accept) {
-				request_accepted = true;
+			if (!accept) {
+				verb_log_dbg_time(
+				   gametime,
+				   "AI Diplomacy: Player(%d)%s denies the %s of player (%d) with diploscore %d%s\n",
+				   static_cast<unsigned int>(pda.other), myts_s.c_str(),
+				   pda.action == Widelands::DiplomacyAction::kInvite ? "invitation" : "join request",
+				   static_cast<unsigned int>(pda.sender), diploscore, other_team_score_str.c_str());
+				game().send_player_diplomacy(pda.other,
+				                             (pda.action == Widelands::DiplomacyAction::kInvite ?
+                                             Widelands::DiplomacyAction::kRefuseInvite :
+                                             Widelands::DiplomacyAction::kRefuseJoin),
+				                             pda.sender);
+			} else {
+				if (planned_action != kNoAction) {
+					verb_log_dbg_time(
+					   gametime,
+					   "AI Diplomacy: Player(%d) replaces plan: old priority: %d, new priority: %d",
+					   static_cast<unsigned int>(mypn), plan_priority, priority);
+					if (planned_action != Widelands::DiplomacyAction::kLeaveTeam) {
+						verb_log_dbg_time(gametime,
+						                  "AI Diplomacy: Player(%d)%s denies the %s of player (%d) with "
+						                  "diploscore %d%s\n",
+						                  static_cast<unsigned int>(mypn), myts_s.c_str(),
+						                  planned_action == Widelands::DiplomacyAction::kAcceptInvite ?
+                                       "invitation" :
+                                       "join request",
+						                  static_cast<unsigned int>(planned_opn), planned_other_score,
+						                  planned_log_append_text.c_str());
+						game().send_player_diplomacy(
+						   mypn,
+						   (planned_action == Widelands::DiplomacyAction::kAcceptInvite ?
```

Comment:
same here I could not find an assignment further up where this could have resulted